GitHubContentUtility/Operations/BlobContentWriter.cs

```diff
@@ -0,0 +1,163 @@
+﻿// ------------------------------------------------------------------------------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------------------------------------------------------------------------------
+
+using Octokit;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using GitHubContentUtility.Common;
+using GitHubContentUtility.Services;
+
+namespace GitHubContentUtility.Operations
+{
+    /// <summary>
+    /// Provides methods for writing contents to a specific blob in a GitHub repository.
+    /// </summary>
+    public class BlobContentWriter
+    {
+        /// <summary>
+        /// Writes contents to a specified blob in the specified GitHub repository.
+        /// </summary>
+        /// <param name="appConfig">The application configuration object which contains values
+        /// for connecting to the specified GitHub repository.</param>
+        /// <param name="privateKey"> The RSA private key of a registered GitHub app installed in the specified repository.</param>
+        /// <returns>A task.</returns>
+        public async Task WriteToRepositoryAsync(ApplicationConfig appConfig, string privateKey)
```

Comment:
do you need a null check on this?

Reply:
Null checking is done here in this class, where the argument is being used.

https://github.com/microsoftgraph/msgraph-github-content-utility/blob/341464d63fbfcc36f8ee04b6d5608bf8b3c26bbe/GitHubContentUtility/Services/GitHubAuthService.cs#L29

Reply:
A general rule of thumb is to null-check all the arguments in all public methods, because by marking it public, it becomes open to be used anywhere other than your current code.

Consider marking the method `internal` or `private` whenever possible. After marking it as such, if you are sure that all code paths leading to this method are covered as you mention here, then you can skip null check.

I am so looking forward to the null checking feature in C#9, so we can simply put an exclamation mark to cover these: https://github.com/dotnet/csharplang/blob/master/proposals/null-arg-checking.md

Reply:
That would so cool